preprocess/preprocess.go

```diff
@@ -96,11 +99,85 @@ func ExpandJSON(block *gabs.Container, vm *goja.Runtime) {
 	log.Debug().Str("Processed JSON block", block.String()).Msg("")
 }
 
-func LoadElfEnv(l2path string) {
+func SearchL2ConfigEnv(dir string) (string, error) {
+	parentDir := filepath.Dir(dir)
+	for parentDir != string(filepath.Separator) {
+		l2ConfigPath := filepath.Join(parentDir, "l2config.env")
+		_, err := os.Stat(l2ConfigPath)
+		if err == nil {
+			return l2ConfigPath, nil // Found the l2config.env file
+		}
+		parentDir = filepath.Dir(parentDir)
+	}
+	return "", errors.New("Didn't find l2config.env in the API directory")
+}
+
+func LoadEnvFile(l2path string) {
+	envFileName := filepath.Base(l2path)
 	err := godotenv.Load(l2path)
 	if err != nil {
-		log.Info().Str("Type", "Preprocess").Msg("Didn't find l2.env in the API directory")
+		log.Info().Str("Type", "Preprocess").Msg("Didn't find "+ envFileName +" in the API directory")
+	}
+}
+
+func LoadEnvironments(dir string) {
+	l2ConfigPath, err := SearchL2ConfigEnv(dir)
+	if err != nil {
+		log.Info().Str("Type", "Preprocess").Msg(err.Error())
+	} else {
+		LoadEnvFile(l2ConfigPath) // Loads global variables from l2config.env
+	}
+	LoadEnvFile(path.Join(dir, "l2.env")) // Overwrites the global variables if declared again in l2.env
+}
+
+func readFile(filename string) (envMap map[string]string, err error) {
+	file, err := os.Open(filename)
+	if err != nil {
+		return
+	}
+	defer file.Close()
+	return godotenv.Parse(file)
+}
+
+func populateEnvMap(envMap map[string]map[string]interface{}, envPath string, source string) error {
+	envs, err := readFile(envPath)
+	if err != nil {
+		return err
+	}
+	for key, value := range envs {
+		variable := make(map[string]interface{})
```

Comment:
8 spaces? use the `gofumpt` or some other thing in the makefile to format.

Reply:
You can run both of these and make any mods:

```

lint:
	golangci-lint run --disable-all -E revive ./...

gofumpt:
	gofumpt -w .

```

Reply:
ok